### PR TITLE
Refactor logger hooks

### PR DESCRIPTION
**Pull request by daavoo**
This P.R. fixes some inconsistencies regarding logger hooks moving the shared logic to the base class.

### CODE REVIEW

**Comment by hellock** [comment]
ping @ycxioooong

File: mmcv/runner/hooks/logger/base.py [hunk 1]
```diff
@@ -31,6 +35,104 @@ def __init__(self,
...
+        if isinstance(val, numbers.Number):
+            return True
+        elif include_np and isinstance(val, np.ndarray) and val.ndim == 0:
+            return True
+        elif include_torch and isinstance(val, torch.Tensor) and len(val) == 1:
+            return True
+        else:
+            return False
+
+    def get_mode(self, runner):
```
Comment (ycxioooong) [inline mmcv/runner/hooks/logger/base.py:59]: I'm not sure why we use "time" to decide the current mode. It's not very elegant.

Comment (daavoo) [reply mmcv/runner/hooks/logger/base.py:59]: I followed the logic used by text logger hook in master branch

**Comment by hellock** [comment]
ping @ycxioooong

**Review by ycxioooong — APPROVED** [review]
LGTM now; By the way, we have a plan to refactor all the logging system for mmcv in the coming months.

File: mmcv/runner/hooks/logger/base.py [hunk 2]
```diff
@@ -31,6 +35,104 @@ def __init__(self,
...
+            current_iter = runner.inner_iter + 1
+        else:
+            current_iter = runner.iter + 1
+        return current_iter
+
+    def get_step(self, runner):
+        if self.get_mode(runner) == 'val' and self.by_epoch:
+            return self.get_epoch(runner)
+        else:
+            return self.get_iter(runner)
```
Comment (xvjiarui) [inline mmcv/runner/hooks/logger/base.py:95]: Hi @daavoo 
What is the difference between there two func?

**Pull request merged**